### PR TITLE
create internal/config/gateway test

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

I created the test for `internal/config/gateway`.

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/config/gateway.go [hunk 1]
```diff
@@ -17,6 +17,7 @@
 // Package config providers configuration type and load configuration logic
 package config
 
+// Gateway represents the list of configuration for gateway.
```
Comment (hlts2) [inline internal/config/gateway.go:20]:
```suggestion
// Gateway represents the list of configurations for gateway.
```

File: internal/config/gateway_test.go [hunk 2]
```diff
@@ -55,58 +58,188 @@ func TestGateway_Bind(t *testing.T) {
...
+					t.Helper()
+					for k, v := range p {
+						if err := os.Setenv(k, v); err != nil {
+							t.Fatal(err)
+						}
+					}
+				},
+				afterFunc: func(t *testing.T) {
+					t.Helper()
+					for k, _ := range p {
```
[the remark this answers is not in the document]
Comment (hlts2) [reply internal/config/gateway_test.go:201]: should we change to the following implementation? 🤔 
```suggestion
					for k := range p {
```

File: internal/config/gateway_test.go [hunk 3]
```diff
@@ -128,6 +261,7 @@ func TestGateway_Bind(t *testing.T) {
 			if err := test.checkFunc(test.want, got); err != nil {
 				tt.Errorf("error = %v", err)
 			}
+
```
[the remark this answers is not in the document]
Comment (hlts2) [reply internal/config/gateway_test.go:264]: Should we remove this line? 🤔

[the remark this answers is not in the document]
Comment (vankichi) [reply internal/config/gateway_test.go:264]:
```suggestion
```

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by kevindiu — APPROVED** [review]
LGTM

**Pull request merged**